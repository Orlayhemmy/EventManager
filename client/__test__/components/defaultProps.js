export const props = {
  userState: {
    user: {
      isAdmin: true
    },
    isAuth: true
  },
  location: {
    pathname: '/'
  },
  center: {
    status: '',
    message: ''
  }
};

export const center = {
  status: 201,
  message: ''
};

export const bookedEvent = [
  { id: 1, eventTitle: 'Pizza Party', bookedDate: [], isApproved: true },
  { id: 2, eventTitle: 'Wedding Ceremony', bookedDate: [], isApproved: true  },
  { id: 3, eventTitle: 'Seminar', bookedDate: [], isApproved: false  }
];

export const logout = jest.fn();
export const onChange = jest.fn();
