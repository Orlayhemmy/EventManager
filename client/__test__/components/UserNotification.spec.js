import React from 'react';
import renderer from 'react-test-renderer';
import Notifications from '../../components/Dashboard/Template/Content/UserNotifications';
import { onClick, activity } from './defaultProps';

describe('Home Component', () => {
  const props = {
    activities: activity,
    onClick
  };
  it('should render without throwing error', () => {
    const wrapper = shallow(<Notifications {...props} />);
    console.log(wrapper.instance())
    expect(wrapper).toMatchSnapshot();
  });
  // it('should carousel when value is true', () => {
  //   const wrapper = renderer.create(<Notifications isAuth={props.isAuth} centerState={props.centerState}/>).toJSON();
  //   expect(wrapper).toMatchSnapshot();
  // });
});
