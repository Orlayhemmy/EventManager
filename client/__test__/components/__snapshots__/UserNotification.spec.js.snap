// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render without throwing error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notifications
    activities={
      Object {
        "activities": Array [
          Object {
            "createdAt": "2018-07-07 17:03:36.922+01",
            "description": "come and eat pizza",
            "eventId": 1,
          },
        ],
      }
    }
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row card p-1 mb-1"
        >
          <span>
            <p
              className="activity-font mb-0 p-1"
              id={undefined}
              onClick={[MockFunction]}
            >
              <br />
            </p>
          </span>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "activities",
        "nodeType": "host",
        "props": Object {
          "children": <span>
            <p
              className="activity-font mb-0 p-1"
              id={undefined}
              onClick={[MockFunction]}
            >
              <br />
            </p>
          </span>,
          "className": "row card p-1 mb-1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="activity-font mb-0 p-1"
              id={undefined}
              onClick={[MockFunction]}
            >
              <br />
            </p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                <br />,
                undefined,
              ],
              "className": "activity-font mb-0 p-1",
              "id": undefined,
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              undefined,
            ],
            "type": "p",
          },
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row card p-1 mb-1"
          >
            <span>
              <p
                className="activity-font mb-0 p-1"
                id={undefined}
                onClick={[MockFunction]}
              >
                <br />
              </p>
            </span>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "activities",
          "nodeType": "host",
          "props": Object {
            "children": <span>
              <p
                className="activity-font mb-0 p-1"
                id={undefined}
                onClick={[MockFunction]}
              >
                <br />
              </p>
            </span>,
            "className": "row card p-1 mb-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="activity-font mb-0 p-1"
                id={undefined}
                onClick={[MockFunction]}
              >
                <br />
              </p>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  <br />,
                  undefined,
                ],
                "className": "activity-font mb-0 p-1",
                "id": undefined,
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                undefined,
              ],
              "type": "p",
            },
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
